app/routes/cart/route.tsx

```diff
@@ -1,55 +1,63 @@
-};
+export default function CartPage() {
+    const { cartData, cartTotals, checkout, removeItem, updateItemQuantity } = useCart();
 
-const noop = () => {};
+    if (!cartData) return null;
+
+    if (!cartData.lineItems.length)
+        return (
+            <div className={styles.cart}>
+                <h1 className={styles.cartHeader}>My cart</h1>
+                <div className={styles.emptyCartMessage}>
+                    <div>Cart is empty</div>
+                    <NavLink to={ROUTES.home.to()} className={styles.continueBrowsingLink}>
+                        Continue Browsing
+                    </NavLink>
+                </div>
+            </div>
+        );
 
-export default function CartPage() {
     return (
         <div className={styles.page}>
             <div className={styles.cart}>
                 <h1 className={styles.cartHeader}>My cart</h1>
                 <div className={styles.cartItems}>
-                    <CartItem
-                        item={mockCartItem}
-                        priceBreakdown={{
-                            lineItemPrice: { formattedConvertedAmount: '$5.50' },
-                        }}
-                        onQuantityChange={noop}
-                        onRemove={noop}
-                    />
-                    <CartItem
-                        item={mockCartItem}
-                        priceBreakdown={{
-                            lineItemPrice: { formattedConvertedAmount: '$5.50' },
-                        }}
-                        onQuantityChange={noop}
-                        onRemove={noop}
-                    />
+                    {cartData?.lineItems.map((item) => (
+                        <CartItem
+                            key={item._id}
+                            item={item}
+                            priceBreakdown={findLineItemPriceBreakdown(item, cartTotals)}
+                            onRemove={() => removeItem(item._id!)}
+                            onQuantityChange={(quantity: number) =>
+                                updateItemQuantity({ id: item._id!, quantity })
+                            }
+                        />
+                    ))}
                 </div>
             </div>
             <div className={styles.summary}>
                 <h1 className={styles.summaryHeader}>Order summary</h1>
                 <div className={styles.summarySection}>
                     <div className={styles.summaryRow}>
                         <span>Subtotal</span>
-                        <span>$5.50</span>
+                        <span>{cartTotals?.priceSummary?.subtotal?.formattedConvertedAmount}</span>
                     </div>
                     <div className={styles.summaryRow}>
                         <span>Delivery</span>
-                        <span>FREE</span>
+                        <span>
+                            {cartTotals?.priceSummary?.shipping?.amount === '0.00'
```

Comment:
I think `Number(priceSummary?.shipping?.amount) === 0` would be more reliable. For example, api can suddently change formatting to one number of decimal places.